monai/utils/type_conversion.py

```diff
@@ -0,0 +1,201 @@
+
+    return data
+
+
+def convert_to_numpy(data):
+    """
+    Utility to convert the input data to a numpy array. If passing a dictionary, list or tuple,
+    recursively check every item and convert it to numpy array.
+
+    Args:
+        data: input data can be PyTorch Tensor, numpy array, list, dictionary, int, float, bool, str, etc.
+            will convert Tensor, Numpy array, float, int, bool to numpy arrays, strings and objects keep the original.
+            for dictionary, list or tuple, convert every item to a numpy array if applicable.
+
+    """
+    if isinstance(data, torch.Tensor):
+        data = data.detach().cpu().numpy()
+    elif has_cp and isinstance(data, cp_ndarray):
+        data = cp.asnumpy(data)
+    elif isinstance(data, (float, int, bool)):
+        data = np.asarray(data)
+    elif isinstance(data, dict):
+        return {k: convert_to_numpy(v) for k, v in data.items()}
+    elif isinstance(data, list):
+        return [convert_to_numpy(i) for i in data]
+    elif isinstance(data, tuple):
+        return tuple(convert_to_numpy(i) for i in data)
+
+    if isinstance(data, np.ndarray) and data.ndim > 0:
+        data = np.ascontiguousarray(data)
+
+    return data
+
+
+def convert_data_type(
+    data: Any,
+    output_type: Optional[type] = None,
+    device: Union[torch.device, str, None] = None,
+    dtype: Optional[Union[DtypeLike, torch.dtype]] = None,
+) -> Tuple[NdarrayTensor, type, Optional[torch.device]]:
+    """
+    Convert to `torch.Tensor`/`np.ndarray` from `torch.Tensor`/`np.ndarray`/`float`/`int` etc.
+
+    Args:
+        data: data to be converted
+        output_type: `torch.Tensor` or `np.ndarray` (if blank, unchanged)
+        device: if output is `torch.Tensor`, select device (if blank, unchanged)
+        dtype: dtype of output data. Converted to correct library type (e.g.,
+            `np.float32` is converted to `torch.float32` if output type is `torch.Tensor`).
+            If left blank, it remains unchanged.
+    Returns:
+        modified data, orig_type, orig_device
+    """
+    orig_type = type(data)
+    orig_device = data.device if isinstance(data, torch.Tensor) else None
+
+    output_type = output_type or orig_type
+
+    dtype = get_equivalent_dtype(dtype or get_dtype(data), output_type)
+
+    if output_type is torch.Tensor:
+        if orig_type is not torch.Tensor:
+            data = convert_to_tensor(data)
+        if dtype != data.dtype:
+            data = data.to(dtype)  # type: ignore
+    elif output_type is np.ndarray:
+        if orig_type is not np.ndarray:
+            data = convert_to_numpy(data)
+        if data is not None and dtype != data.dtype:
+            data = data.astype(dtype)  # type: ignore
+
+    if isinstance(data, torch.Tensor) and device is not None:
+        data = data.to(device)
+
+    return data, orig_type, orig_device
+
+
+def convert_same_type(src: Any, dst: NdarrayTensor) -> Tuple[NdarrayTensor, type, Optional[torch.device]]:
```

Comment:
I added this utility and it doesn't have the typing issue. but I'm not sure about the name of the function, do you have any other idea? @rijobro @Nic-Ma @ericspod

Reply:
Hi @wyli ,

Thanks for the quick update.
It's a little bit confusing, but acceptable. Maybe `convert_to_dst_type()` is more clear?

Thanks.